### PR TITLE
[core][compiled-graphs][1/N] Polish APIs: execution_timeout, retrieval_timeout, buffer_size_bytes

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

* Rename `execution_timeout` to `submit_timeout`.
* Rename `retrieval_timeout` to `get_timeout` and update comments.
* Update comments of `buffer_size_bytes`

## Related issue number

https://github.com/ray-project/ray/issues/49051

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/dag/context.py [hunk 1]
```diff
@@ -8,8 +8,8 @@
 _default_context: "Optional[DAGContext]" = None
 _context_lock = threading.Lock()
 
-DEFAULT_EXECUTION_TIMEOUT_S = int(os.environ.get("RAY_DAG_execution_timeout", 10))
-DEFAULT_RETRIEVAL_TIMEOUT_S = int(os.environ.get("RAY_DAG_retrieval_timeout", 10))
+DEFAULT_SUBMIT_TIMEOUT_S = int(os.environ.get("RAY_DAG_submit_timeout", 10))
```
Comment (ruisearch42) [inline python/ray/dag/context.py:11]:
```suggestion
DEFAULT_SUBMIT_TIMEOUT_S = int(os.environ.get("RAY_CGRAPH_submit_timeout", 10))
```

File: python/ray/dag/context.py [hunk 2]
```diff
@@ -8,8 +8,8 @@
...
+DEFAULT_GET_TIMEOUT_S = int(os.environ.get("RAY_DAG_get_timeout", 10))
```
Comment (ruisearch42) [inline python/ray/dag/context.py:12]:
```suggestion
DEFAULT_GET_TIMEOUT_S = int(os.environ.get("RAY_CGRAPH_get_timeout", 10))
```

File: python/ray/dag/context.py [hunk 3]
```diff
@@ -8,8 +8,8 @@
...
 DEFAULT_TEARDOWN_TIMEOUT_S = int(os.environ.get("RAY_DAG_teardown_timeout", 30))
 # Default buffer size is 1MB.
 DEFAULT_BUFFER_SIZE_BYTES = int(os.environ.get("RAY_DAG_buffer_size_bytes", 1e6))
```
Comment (ruisearch42) [inline python/ray/dag/context.py:15]: Ditto, RAY_CGRAPH

**Review by ruisearch42 — APPROVED** [review]

File: python/ray/dag/context.py (hunk 1 above)
Comment (kevin85421) [reply python/ray/dag/context.py]: updated

File: python/ray/dag/context.py (hunk 3 above)
Comment (kevin85421) [reply python/ray/dag/context.py]: updated

**Review by ruisearch42 — APPROVED** [review]

**Pull request merged**